deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -171,14 +171,66 @@ func (c CCIPOCRParams) Validate(selector uint64, feedChainSel uint64, state CCIP
 	return nil
 }
 
-// DefaultOCRParams returns the default OCR parameters for a chain,
-// except for a few values which must be parameterized (passed as arguments).
-func DefaultOCRParams(
-	feedChainSel uint64,
-	tokenInfo map[ccipocr3.UnknownEncodedAddress]pluginconfig.TokenInfo,
-	tokenDataObservers []pluginconfig.TokenDataObserverConfig,
-	commit bool,
-	exec bool,
+type CCIPOCROpts func(params *CCIPOCRParams)
+
+// WithOCRParamOverride can be used if you want to override the default OCR parameters with your custom function.
+func WithOCRParamOverride(override func(params *CCIPOCRParams)) CCIPOCROpts {
+	return func(params *CCIPOCRParams) {
+		if override != nil {
+			override(params)
+		}
+	}
+}
+
+// WithDefaultCommitOffChainConfig can be used to add token info to the existing commit off-chain config. If no commit off-chain config is set, it will be created with default values.
+func WithDefaultCommitOffChainConfig(feedChainSel uint64, tokenInfo map[ccipocr3.UnknownEncodedAddress]pluginconfig.TokenInfo) CCIPOCROpts {
+	return func(params *CCIPOCRParams) {
+		if params.CommitOffChainConfig == nil {
+			params.CommitOffChainConfig = &pluginconfig.CommitOffchainConfig{
+				RemoteGasPriceBatchWriteFrequency:  *config.MustNewDuration(internal.RemoteGasPriceBatchWriteFrequency),
+				TokenPriceBatchWriteFrequency:      *config.MustNewDuration(internal.TokenPriceBatchWriteFrequency),
+				TokenInfo:                          tokenInfo,
+				PriceFeedChainSelector:             ccipocr3.ChainSelector(feedChainSel),
+				NewMsgScanBatchSize:                merklemulti.MaxNumberTreeLeaves,
+				MaxReportTransmissionCheckAttempts: 5,
+				RMNEnabled:                         os.Getenv("ENABLE_RMN") == "true", // only enabled in manual test
```

Comment:
Why not have RMN always enabled and explicitly disable when not needed?

Reply:
discussed offline